### DISCUSSION TITLE
Difference between this and your Net CLI

### DISCUSSION THREAD

**Discussion by Asbjoedt** [mikeebowen/ooxml-validator-vscode#16] [Q&A] [answered]
Hi @mikeebowen 

I came across another repository you have, for OOXMXL validation
https://github.com/mikeebowen/OOXML-Validator

Does this and will this repository have the same functionality as this VSCODE extension?

_I think our needs are primarily to integrate a CLI workflow to do bulk validation of thousands of files, that are submitted to our archive in information packages containing typically 5 years of data from IT systems. The validation information should be output in CSV, JSON or preferably XML._

Sorry if I interacted with you on the wrong repository. Let me know where you want my input to be registered.

**Comment by mikeebowen** [comment, marked as answer]
Hi @Asbjoedt ,

If you want to work on thousands of files then the CLI would be a better fit. The CLI is a wrapper around the Open XML SDK that returns the validation errors as as JSON file and the vscode extension reads that JSON file and presents it in a webview, so all the information available from the extension comes from the CLI. It currently only returns json and then the extension converts it to csv to save a log of the errors. It should be possible to add a flag to return xml instead of json. If you want to make a feature request for that [the CLI repo](https://github.com/mikeebowen/OOXML-Validator/issues) would be the best place to do that. 

Could you tell me more about how you want to use the CLI? Are you planning on writing a wrapper for it that bulk validates a trove of files and then creates a report about validation errors and strict status?

**Reply by Asbjoedt** [reply]
@mikeebowen thanks for replying and thanks for asking to our usercase.

Yes! I am not so well into terms of "wrapper", but from what I gather, this may be the case. We want to have a CLI workflow bulkprocessing the documents folder in submitted Information Packages and return a list of validation errors (and potentially hints (acceptable errors)) based on comparison to a validation XML schema declaring our validation requirements. The errors should be viewable in a simple GUI showing which documents have which validation errors. Potentially this GUI should be merged into our general purpose validation tool, that we have developed and maintain in our organisation, but this is not a sure decision at the moment.

Below you can see a list of validation requirements, that we are currenlty discussing. As you may see, our idea is that your validator can help us check for requirement 1, if check for strict conformance is adopted.

1. **Standard conformance:** Spreadsheets must be converted to Office Open XML in accordance with the standard ISO/IEC 29500-1:2016. The file format extension must be .xlsx. The file format must be conformant with the document conformance class Office Open XML Strict and SML Strict.
2. **Password protection:** Spreadsheets must not be password protected of any sort e.g. sheets, structure or cells.
3. **Template file formats:** Templates in the file formats OTS, XLT, XLTX, XLTM must be converted to XLSX Strict file format according to requirement no. 1.
4. **Links to other spreadsheets:** Cells with links to data in other spreadsheets must be converted to one of the following: Values, Values and number formating or Values and source formatting.
5. **External datasources:** External data sources must be removed. The current cell values must be kept.
6. **RealTimeData functions:** RealTimeData (RTD) functions must be removed. Current cell values must be kept.
7. **Embedded documents:** Embedded documents e.g. Word and PowerPoint must be converted to their corresponding and approved file formats. 
8. **Embedded media:** Embedded media e.g. audio, images and video must be converted to their corresponding and approved file formats.
